ssh-agent.c

```diff
@@ -2383,18 +2401,22 @@ main(int ac, char **av)
 		strlcpy(socket_name, agentsocket, sizeof socket_name);
 	}
 
+	closefrom(sock == -1 ? STDERR_FILENO + 1 : sock + 1);
```

Comment:
Why is this introduced in this patch?  it seems like this change (closing extra file descriptors) is orthogonal to the purpose of the patch.